### ISSUE TITLE
Should we change the API?

### ISSUE THREAD

**Issue by rowanmanning** [Financial-Times/origami-service#17]
Currently the whole app creation/listening is combined into one call which returns a promise that resolves to the app:

```js
origamiService(options).then(app => {
    // add routes etc
});
// app has started automatically
```

Does something like this feel more Express-like and familiar? Should we change it?

```js
const app = origamiService(options);

// add routes etc

app.listen().then(() => {
    console.log('app started');
});
```

**Comment by onishiweb** [comment]
The latter certainly feels a lot more familiar being the express convention. But then I've only built a single Express app so far. 

If the service has a lot of routes, the code in the current method could get a bit bloated and hard to follow being all in the config object, so maybe the latter would be better?

**Comment by JakeChampion** [comment]
Agree with Adam. We should return the app so that the user can decide when they listen and if they want to make anymore changes to the app prior to listening.
